web3.js/src/util/send-and-confirm-transaction.ts

```diff
@@ -27,23 +47,29 @@ export async function sendAndConfirmTransaction(
     maxRetries: options.maxRetries,
   };
 
+  let status;
+
   const signature = await connection.sendTransaction(
     transaction,
     signers,
     sendOptions,
   );
 
-  const status = (
+  const signatureBlockhash: TransactionSignatureBlockhash = {
+    signature: signature,
+    blockhash: transaction.recentBlockhash,
+    lastValidBlockHeight: transaction.lastValidBlockHeight,
+  };
+
+  status = (
     await connection.confirmTransaction(
-      signature,
+      signatureBlockhash,
       options && options.commitment,
     )
   ).value;
```

Comment:
Think of how much easier this will read if you change everything to use the word ‘strategy,’ as suggested above.

Reply:
Yeah, I agree. It's getting pretty messy to look at right now.